plugin.video.newson/default.py

```diff
@@ -0,0 +1,270 @@
+# Plugin Info
+ADDON_ID      = 'plugin.video.newson'
+REAL_SETTINGS = xbmcaddon.Addon(id=ADDON_ID)
+ADDON_NAME    = REAL_SETTINGS.getAddonInfo('name')
+SETTINGS_LOC  = REAL_SETTINGS.getAddonInfo('profile')
+ADDON_PATH    = REAL_SETTINGS.getAddonInfo('path').decode('utf-8')
+ADDON_VERSION = REAL_SETTINGS.getAddonInfo('version')
+ICON          = REAL_SETTINGS.getAddonInfo('icon')
+FANART        = REAL_SETTINGS.getAddonInfo('fanart')
+NEWSART       = os.path.join(ADDON_PATH,'resources','images','newscast.jpg')
+CLIPART       = os.path.join(ADDON_PATH,'resources','images','videoclips.jpg')
+
+## GLOBALS ##
+TIMEOUT     = 30
+DEBUG       = True
+BASE_URL    = 'http://watchnewson.com/'
+BASE_API    = 'http://watchnewson.com/api/linear/channels'
+USER_REGION = 'US'
+MENU        = [("Newscasts"  , '0', 0, False, {"thumb":NEWSART,"poster":NEWSART,"fanart":FANART,"icon":ICON,"logo":ICON}),
+               ("Video Clips", '2', 2, False, {"thumb":CLIPART,"poster":CLIPART,"fanart":FANART,"icon":ICON,"logo":ICON})]
+
+def log(msg, level=xbmc.LOGDEBUG):
+    if DEBUG == True:
+        if level == xbmc.LOGERROR:
+            msg += ' ,' + traceback.format_exc()
+        xbmc.log(ADDON_ID + '-' + ADDON_VERSION + '-' + stringify(msg), level)
+        
+def stringify(string):
+    if isinstance(string, list):
+        string = stringify(string[0])
+    elif isinstance(string, (int, float, long, complex, bool)):
+        string = str(string) 
+    if isinstance(string, basestring):
+        if not isinstance(string, unicode):
+            string = unicode(string, 'utf-8')
+        elif isinstance(string, unicode):
+            string = string.encode('ascii', 'ignore')
+        else:
+            string = string.encode('utf-8', 'ignore')
+    return string
+
+def getParams():
+    param=[]
+    if len(sys.argv[2])>=2:
+        params=sys.argv[2]
+        cleanedparams=params.replace('?','')
+        if (params[len(params)-1]=='/'):
+            params=params[0:len(params)-2]
+        pairsofparams=cleanedparams.split('&')
+        param={}
+        for i in range(len(pairsofparams)):
+            splitparams={}
+            splitparams=pairsofparams[i].split('=')
+            if (len(splitparams))==2:
+                param[splitparams[0]]=splitparams[1]
+    return param
+                 
+socket.setdefaulttimeout(TIMEOUT)
+
+class NewsOn():
+    def __init__(self):
+        log('__init__')
+        self.cache = SimpleCache()
+        self.stateMenu = self.getStates()
+
+        
+    def openURL(self, url):
+        try:
+            cacheResponce = self.cache.get(ADDON_NAME + '.openURL, url = %s'%url)
+            if not cacheResponce:
+                request = urllib2.Request(url)
+                request.add_header('User-Agent','Mozilla/5.0 (Windows; U; MSIE 9.0; Windows NT 9.0; en-US)')
+                page = urllib2.urlopen(request, timeout = TIMEOUT)
+                log(page.headers['content-type'])
+                results = json.load(page)
```

Comment:
I've been periodically checking their return headers for changes. I believe the site is in the middle of making changes. I'll check again for compression.